### PR TITLE
Test light client circuit on mockda with mockzkvm

### LINKED ISSUE #1374: `ligth_client_prover::run_circuit()` MockDA and MockZkVm tests
basically unit tests to test out the light client circuit.

we can also do the same for batch proving, however, we opted to e2e tests always.

### PR DESCRIPTION
**Pull request by ercecan**
# Description
Test slight client circuit on mockda with mockzkvm

## Linked Issues
- Fixes #1374

### CODE REVIEW

File: crates/light-client-prover/src/circuit.rs [hunk 1]
```diff
@@ -37,7 +38,7 @@ pub fn run_circuit<DaV: DaVerifier, G: ZkvmGuest>(
 
     let block_updates = da_verifier
         .verify_header_chain(&previous_light_client_proof_output, &input.da_block_header)
-        .expect("Failed to verify DA header chain");
+        .map_err(|_| LightClientVerificationError::HeaderChainVerificationFailed)?;
```
Comment (eyusufatik) [inline crates/light-client-prover/src/circuit.rs:44]: dont forget to update the actual guests

File: crates/light-client-prover/src/circuit.rs [hunk 2]
```diff
@@ -99,8 +100,16 @@ pub fn run_circuit<DaV: DaVerifier, G: ZkvmGuest>(
...
+                            match G::verify_and_extract_output(
+                                &journal,
+                                &batch_proof_method_id.into(),
+                            ) {
+                                Ok(output) => output,
+                                Err(e) => {
+                                    println!("error: {:?}", e);
+                                    continue;
+                                }
+                            };
```
Comment (eyusufatik) [inline crates/light-client-prover/src/circuit.rs]: just use ? and update guests to panic on errors

File: crates/light-client-prover/src/tests/mod.rs [hunk 3]
```diff
@@ -0,0 +1,339 @@
...
+    let blob_4 = create_mock_blob([4u8; 32], [5u8; 32], 5, true);
+
+    let block_header_2 = MockBlockHeader::from_height(2);
+
+    let mock_output_1_serialized = create_prev_lcp_serialized(output_1, true);
+
+    let input_2 = LightClientCircuitInput::<MockDaSpec> {
+        previous_light_client_proof_journal: Some(mock_output_1_serialized),
+        da_block_header: block_header_2,
+        da_data: vec![blob_3, blob_4],
```
Comment (eyusufatik) [inline crates/light-client-prover/src/tests/mod.rs:55]: i thought there could be only 1 blob per mock da block?

File: crates/light-client-prover/src/circuit.rs
Comment (eyusufatik) [inline crates/light-client-prover/src/circuit.rs:1]: Got invalid previous light client proof

can you also make ^ an enum member?

File: crates/light-client-prover/src/tests/mod.rs [hunk 4]
```diff
@@ -0,0 +1,339 @@
...
+        inclusion_proof: [1u8; 32],
+        completeness_proof: (),
+        l2_genesis_state_root: None,
+        batch_proof_method_id: light_client_proof_method_id,
+        batch_prover_da_pub_key: [9; 32].to_vec(),
+    };
+
+    guest.input = borsh::to_vec(&input_2).unwrap();
+
+    let _ = run_circuit(da_verifier, &guest).unwrap();
```
Comment (eyusufatik) [inline crates/light-client-prover/src/tests/mod.rs]: err comparison

**Review by eyusufatik — CHANGES_REQUESTED** [review]

File: crates/light-client-prover/src/tests/mod.rs (hunk 3 above)
Comment (ercecan) [reply crates/light-client-prover/src/tests/mod.rs:55]: Yes, but we are not passing da blocks, we are passing blobs

File: crates/light-client-prover/src/circuit.rs (hunk 2 above)
Comment (ercecan) [reply crates/light-client-prover/src/circuit.rs]: This is not a place to error out, we should continue if the batch proof read from da is unverifiable. Was that not the case?

File: crates/light-client-prover/src/tests/mod.rs (hunk 3 above)
Comment (eyusufatik) [reply crates/light-client-prover/src/tests/mod.rs:55]: and mockda allways verifies?

File: crates/light-client-prover/src/circuit.rs (hunk 2 above)
Comment (eyusufatik) [reply crates/light-client-prover/src/circuit.rs]: ah sorry i thought it was lcpp

Comment (eyusufatik) [reply crates/light-client-prover/src/circuit.rs]: remove println

File: crates/light-client-prover/src/tests/mod.rs (hunk 3 above)
Comment (ercecan) [reply crates/light-client-prover/src/tests/mod.rs:55]: yes, is that wrong behavior for mock da?

**Review by eyusufatik — APPROVED** [review]

**Pull request merged**